src/pymor/algorithms/newton.py

```diff
@@ -124,6 +128,15 @@ def newton(operator, rhs, initial_guess=None, mu=None, error_norm=None, least_sq
             correction = jacobian.apply_inverse(residual, least_squares=least_squares)
         except InversionError:
             raise NewtonError('Could not invert jacobian')
+        if line_search_method is not None:
+            if line_search_method == 'armijo':
+                def res(x):
+                    residual_vec = rhs - operator.apply(x, mu=mu)
+                    return residual_vec.l2_norm()[0] if error_norm is None else error_norm(residual_vec)[0]
+                grad = 2.0 * jacobian.apply(residual) if error_norm is None else None
+                relax = armijo(res, U, correction, grad=grad)
+            else:
+                raise NewtonError('Unknown line search method')
```

Comment:
I would not use `NewtonError` here, which should be a sign of failed convergence. I would also be fine with `assert False` or `ValueError`.